File: Jasmine.FastenerDepartment.Wpf/Jasmine.FastenerDepartment.Application/Models/Synchronization/SynchronizationProduct.cs
using Jasmine.FastenerDepartment.Domain.MeasurementUnits.Models;
using Jasmine.FastenerDepartment.Domain.PriceTags.Models;

namespace Jasmine.FastenerDepartment.Application.Models.Synchronization;

/// <summary>
/// Synchronization product.
/// </summary>
public class SynchronizationProduct
{
    /// <summary>
    /// Identifier.
    /// </summary>
    public Guid Id { get; set; }

    /// <summary>
    /// Created date.
    /// </summary>
    public DateTime CreatedDate { get; set; }

    /// <summary>
    /// Modified date.
    /// </summary>
    public DateTime ModifiedDate { get; set; }

    /// <summary>
    /// Number.
    /// </summary>
    public int Number { get; set; }

    /// <summary>
    /// Name.
    /// </summary>
    public string Name { get; set; }

    /// <summary>
    /// Price.
    /// </summary>
    public decimal Price { get; set; }

    /// <summary>
    /// Is deleted.
    /// </summary>
    public bool IsDeleted { get; set; }

    /// <summary>
    /// Is needed to order.
    /// </summary>
    public bool IsNeededToOrder { get; set; }

    /// <summary>
    /// Is needed to print.
    /// </summary>
    public bool IsNeededToPrint { get; set; }

    /// <summary>
    /// Price tag code.
    /// </summary>
    public PriceTagCode PriceTagCode { get; set; }

    /// <summary>
    /// Measurement unit code.
    /// </summary>
    public MeasurementUnitCode MeasurementUnitCode { get; set; }
}

File: JasmineFastenerDepartment.Kotlin/app/src/main/java/com/szaytsev/jasminefastenerdepartment/network/models/SynchronizationProduct.kt
package com.szaytsev.jasminefastenerdepartment.network.models

import com.szaytsev.jasminefastenerdepartment.data.models.ProductMeasurementUnitCode
import com.szaytsev.jasminefastenerdepartment.data.models.ProductPriceTagCode
import com.szaytsev.jasminefastenerdepartment.data.serializers.PriceTagCodeSerializer
import com.szaytsev.jasminefastenerdepartment.data.serializers.ProductMeasurementUnitCodeSerializer
import com.szaytsev.jasminefastenerdepartment.data.serializers.ZoneDateTimeSerializer
import kotlinx.serialization.Serializable
import java.time.ZonedDateTime

@Serializable
data class SynchronizationProduct(
    val id: String,
    val number: Int,
    val name: String,
    val price: Double,
    val isNeededToOrder: Boolean,
    val isNeededToPrint: Boolean,
    val isDeleted: Boolean,

    @Serializable(PriceTagCodeSerializer::class)
    val priceTagCode: ProductPriceTagCode,

    @Serializable(ProductMeasurementUnitCodeSerializer::class)
    val measurementUnitCode: ProductMeasurementUnitCode,

    @Serializable(with = ZoneDateTimeSerializer::class)
    val createdDate: ZonedDateTime,

    @Serializable(with = ZoneDateTimeSerializer::class)
    val modifiedDate: ZonedDateTime
)